modules/mobile/app/services/mobile/v0/appointments/proxy.rb

```diff
@@ -47,6 +47,20 @@ def put_cancel_appointment(params)
           handle_cancel_error(e, params)
         end
 
+        def get_appointment_request(id)
+          vaos_appointment_requests_service.get_request(id)[:data]
+        end
+
+        def put_cancel_appointment_request(id, request)
+          parsed_request = request.as_json['table']
+          parsed_request['status'] = 'Cancelled'
+          parsed_request['appointment_request_detail_code'] = ['DETCODE8']
```

Comment:
Did you get DETCODE8 from the VAOS code somewhere? How'd you determine this is the one to use?

Reply:
from the VAOS spec appointment request factory. When making a cancellation request it updates the status and detail code to DETCODE8. I believe that code essentially means, "cancelled by user"

Reply:
It is, but according to code I wrote here: modules/mobile/app/models/mobile/v0/templates/base_appointment.rb
there's another cancelled by user case. DETCODE22. I believe they all have a status detail that gives more info, but I don't recall what they were. If this is what they use in VAOS, I'm sure it's fine.

Reply:
oh yeah, I didn't see that. The returned status detail for DETCODE8 is `"provider_message":"Cancelled by Veteran - Pending",
                  "veteran_message":"Your request to cancel this appointment has been sent.",` which seems correct.